docs: Update API versioning doc (#4311)
acts-project/acts: docs/versioning.rst

requests to track the type of changes that have been merged and ensure that the
version number is increased correctly.

> Since ACTS is still under active development, so we create **major** versions
> that break API relatively frequently. We try to limit breaking changes to about
> once a month.

Reviewer: ```suggestion
Since ACTS is still under active development, we create **major** versions
that break the API relatively frequently. We try to limit breaking changes to about
once a month.
```

Public API
----------

> The API surface of ACTS is very large. Everything at the top-level in the ``Acts`` is considered public.

Reviewer 2: ```suggestion
The API surface of ACTS is very large. Everything at the top-level in the ``Acts``-namespace is considered public.
```

.. note::

>     If a private a symbol is part of an otherwise public API, this should be

Reviewer 2: Maybe we should define what a private symbol is. You refer to `Acts::detail` here, right?

requests to track the type of changes that have been merged and ensure that the
version number is increased correctly.

Since ACTS is still under active development, we create **major** versions
that break the API relatively frequently. We try to limit breaking changes to about
once a month.

Public API
----------

The API surface of ACTS is very large. Everything at the top-level in the ``Acts``-namespace is considered public.

.. note::

>     If a private a symbol, e.g. in a the `Acts::detail` or `Acts::Experimental` namespaces
>     is part of an otherwise public API, this should be **considered a bug**, which we'd
>     appreciate being reported as an issue.

Reviewer: ```suggestion
    If a private symbol, e.g. in the `Acts::detail` or `Acts::Experimental` namespaces,
    is part of an otherwise public API, this should be **considered a bug**, which we
    would appreciate being reported as an issue.
```
